packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -299,6 +299,47 @@ - (void)handleCreateMethodCall:(FlutterMethodCall *)call
   });
 }
 
+// Returns number value if provided and positive, or nil.
+// Used to parse values like framerates and bitrates, that are positive by nature.
+// nil allows to ignore unsupported values.
++ (NSNumber *)positiveOrNil:(id)number error:(NSError **)error {
+  if (!number || [number isEqual:[NSNull null]]) {
+    return nil;
+  }
+
+  if (![number isKindOfClass:[NSNumber class]]) {
+    if (error) {
+      *error = [NSError errorWithDomain:@"ArgumentError"
+                                   code:0
+                               userInfo:@{NSLocalizedDescriptionKey : @"should be a number"}];
+    }
+    return nil;
+  }
+
+  NSNumber *value = (NSNumber *)number;
```

Comment:
Nit: it seems very confusing that "value" is the name of the thing when typed as a number, while "number" is the name of the same thing when typed as a generic object value. Let's swap the names of the two variables.

Reply:
names switched